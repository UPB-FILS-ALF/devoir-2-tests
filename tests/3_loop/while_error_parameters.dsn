& while error parameters
SET: %a 10

WHILE: %a 10    30
IDIV: %a 3
COLOR:  100 100 0 blue
COLOR:  pen 255 0 100
RECTANGLE: 100 200 200 100
LINE:  600 100 location
REPEAT

IF: %a
COLOR:  fill 255 0 0
COLOR:  pen 0 0 300
ELLIPSE: 150 50 400 400 
END