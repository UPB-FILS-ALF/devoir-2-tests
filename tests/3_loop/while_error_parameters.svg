<?xml version="1.0" encoding="UTF-8" ?>
<svg xmlns="http://www.w3.org/2000/svg" version="1.1" width="1024" height="768">
	<!-- a = 10 -->
	<!-- a = a / 3 : 3 -->
	<!-- pen 255, 0, 100 -->
	<polygon points="100,200 200,200 200,100 100,100" stroke="rgb(255,0,100)"  stroke-width="1"  fill="rgb(255,255,255)"/>
	<line x1="0" y1="0" x2="600" y2="100" stroke="rgb(255,0,100)"  stroke-width="1" />
	<!-- if a: 3 -->
		<!-- fill 255, 0, 0 -->
		<!-- pen 0, 0, 300 -->
		<ellipse cx="150" cy="50" rx="400" ry="400" stroke="rgb(0,0,300)"  stroke-width="1" fill="rgb(255,0,0)"/>
		<!-- end -->
</svg>
