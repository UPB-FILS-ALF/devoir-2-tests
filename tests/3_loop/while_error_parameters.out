ERROR LINE (4): WHILE has 1 parameters, you wrote 3
ERROR LINE (6): COLOR parameter 1 needs to be one of (pen, fill), you wrote 100
ERROR LINE (10): REPEAT and no LOOP
